app/src/main/java/eu/kanade/tachiyomi/data/related/RelatedUpdateService.kt

```diff
@@ -123,6 +138,13 @@ class RelatedUpdateService(
         val dataToInsert = mutableListOf<MangaRelatedImpl>()
         for (key in relatedPageResult.keys()) {
 
+            // Check if the service is currently running
+            if (job?.isCancelled!!) {
+                cancelProgressNotification()
+                showResultNotification(true)
+                return
```

Comment:
Does this work now?  I couldn't figure out why this would not exit the loop when cancelled and the library one does.  Maybe cause the library one is making network calls?  dunno but thats why the cancel was removed in last build

Reply:
Actually i think this is why it was not working.

The method needs wrapped with  withContext(Dispatchers.IO) {

}

then you shouldnt need this or the job

Reply:
Changed it to this, but still need to have the if inside to check if the loop should end early.